sdk/src/main/java/com/kin/ecosystem/Kin.java

```diff
@@ -192,43 +197,63 @@ public static void login(@NonNull String jwt, KinCallback<Void> loginCallback) {
 	private static void internalLogin(@NonNull final String jwt, final KinCallback<Void> loginCallback) {
 		try {
 			checkInstanceNotNull();
-			if(!AuthRepository.getInstance().isSameUser(jwt)) {
-				logout();
+			@UserLoginState int loginState = AuthRepository.getInstance().getUserLoginState(jwt);
```

Comment:
internal login can be called more than once - consider setting isAccountLoggedIn = false in the beginning of this method

Reply:
consider also changing isAccountLoggedIn to atomic boolean